### ISSUE TITLE
Add optional volumes and volumeMounts to Helm chart

### ISSUE THREAD

**Issue by brunocous** [actions-runner-controller/actions-runner-controller#950]
**Is your feature request related to a problem? Please describe.**
I am using the secret store CSI driver (https://github.com/kubernetes-sigs/secrets-store-csi-driver) to populate pods with secrets from a third party secret store. In order to sync it with Kubernetes secrets, I need to mount a volume to the target pod (see https://secrets-store-csi-driver.sigs.k8s.io/topics/sync-as-kubernetes-secret.html). The Helm chart currently does not provide me an easy way to quickly add an extra`volume` and `volumeMount`. Some other popular charts offer this flexibility already (look at external-dns, neo4j, nginx, etc).

In the future, a user might want to add additional volumes for other purposes.

**Describe the solution you'd like**
Add 2 new fields in `values.yaml` that will add additional volumes and volume mounts to the deployment yaml (https://github.com/actions-runner-controller/actions-runner-controller/blob/master/charts/actions-runner-controller/templates/deployment.yaml). 
````
## specify additional volumes to mount in the core container, this can be used
## to specify additional storage of material or to inject files from ConfigMaps
## into the running container
additionalVolumes: []

## specify where the additional volumes are mounted in the core container
additionalVolumeMounts: []
````
In the `deployment.yaml` template, it suffices to add:
````
        volumeMounts:
        - mountPath: "/etc/actions-runner-controller"
          name: secret
          readOnly: true
        - mountPath: /tmp
          name: tmp
        - mountPath: /tmp/k8s-webhook-server/serving-certs
          name: cert
          readOnly: true
      {{- if .Values.additionalVolumeMounts }}
{{ toYaml .Values.additionalVolumeMounts | indent 8}}
      {{- end }}
````
and
````
      volumes:
      - name: secret
        secret:
          secretName: {{ include "actions-runner-controller.secretName" . }}
      - name: cert
        secret:
          defaultMode: 420
          secretName: {{ include "actions-runner-controller.servingCertName" . }}
      - name: tmp
        emptyDir: {}
        {{- if .Values.additionalVolumes }}
{{ toYaml .Values.additionalVolumes | indent 8}}
        {{- end }}
````

**Describe alternatives you've considered**
Patching an existing deployment with the additional volumes. 
Or specifically for my use case: not using the secret store csi driver, and manually adding the kubernetes secret. Or use other ways to get secrets into a k8s namespace from a third party provider.

**Comment by mumoshu** [comment]
@brunocous Hey! Thanks a lot for writing this up. This sounds like a valid feature request to our chart. I think I've seen this `additionalVolumes` and `additionalVolumeMounts` pair many times in various Helm stable charts so the values API seems good too.

Would you mind submitting a PR for this, if you have any time/experience writing charts?

**Comment by brunocous** [comment]
Fixed in #952
